### PR TITLE
[TLUNIZIN-747] Create API that returns actual Course/Term data from ESB/MPathways

### PR DESCRIPTION
**Pull request by dovek**
DUE: 10/27/2015
@dlhaines 
@pushyamig

### CODE REVIEW

File: src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java [hunk 1]
```diff
@@ -99,6 +87,29 @@
 	protected static Properties appExtSecurePropertiesFile=null;
 	protected static Properties appExtPropertiesFile=null;	
 
+	private static final HashMap<String,String> apiListRegexWithDebugMsg = new HashMap<String,String>(){
+		private static final long serialVersionUID = -1389517682290891890L;
```
Comment (dlhaines) [inline src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java]: Why is the serialVersionUID in the hashmap initialization?

File: src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java [hunk 2]
```diff
@@ -262,13 +273,41 @@ private void apiConnectionLogic(HttpServletRequest request, HttpServletResponse
...
+		String mpathwaysInstructor = request.getParameter(PARAMETER_INSTRUCTOR);
+		String mpathwaysTermId = request.getParameter(PARAMETER_TERMID);
+		if(mpathwaysInstructor == null || 
+				mpathwaysTermId == null){
+			response.setStatus(400);
+			wrappedResult = new WAPIResultWrapper(400, "Parameter missing in Instructors request", new JSONObject());
+		}
+		else{
+			if(appExtSecurePropertiesFile!=null) {
+				HashMap<String, String> value = new HashMap<String, String>();
```
Comment (dlhaines) [inline src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java]: A name more specific than 'value' would be nice.

File: src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java [hunk 3]
```diff
@@ -262,13 +273,41 @@ private void apiConnectionLogic(HttpServletRequest request, HttpServletResponse
...
+				value.put("tokenServer", appExtSecurePropertiesFile.getProperty(SectionUtilityToolFilter.ESB_TOKEN_SERVER));
+				value.put("apiPrefix", appExtSecurePropertiesFile.getProperty(SectionUtilityToolFilter.ESB_PREFIX));
+				value.put("key", appExtSecurePropertiesFile.getProperty(SectionUtilityToolFilter.ESB_KEY));
+				value.put("secret", appExtSecurePropertiesFile.getProperty(SectionUtilityToolFilter.ESB_SECRET));
+				WAPI wapi = new WAPI(value);
```
Comment (dlhaines) [inline src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java]: Why not make the WAPI instance once and then reuse it?  Thread issues?

File: src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java [hunk 4]
```diff
@@ -262,13 +273,41 @@ private void apiConnectionLogic(HttpServletRequest request, HttpServletResponse
...
+				try {
+					wrappedResult = wapi.getRequest(wapi.getApiPrefix() + mpathwaysInstructor + "/Terms/" + mpathwaysTermId + "/Classes");
+				} catch (UnirestException e) {
+					e.printStackTrace();
```
Comment (pushyamig) [inline src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java]: Log Error than simply printing stackTrace

File: src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java [hunk 5]
```diff
@@ -262,13 +273,41 @@ private void apiConnectionLogic(HttpServletRequest request, HttpServletResponse
...
 	}
 
+	private void mpathwaysCall(HttpServletRequest request, HttpServletResponse response, PrintWriter out){
+		WAPIResultWrapper wrappedResult = null;
+		String mpathwaysInstructor = request.getParameter(PARAMETER_INSTRUCTOR);
+		String mpathwaysTermId = request.getParameter(PARAMETER_TERMID);
+		if(mpathwaysInstructor == null || 
+				mpathwaysTermId == null){
+			response.setStatus(400);
+			wrappedResult = new WAPIResultWrapper(400, "Parameter missing in Instructors request", new JSONObject());
```
Comment (pushyamig) [inline src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java]: You can Log this error condition

File: src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java [hunk 6]
```diff
@@ -262,13 +273,41 @@ private void apiConnectionLogic(HttpServletRequest request, HttpServletResponse
...
+		}
```
Comment (pushyamig) [inline src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java]: you can simply return after checking first `if` for readability

File: src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java (hunk 1 above)
Comment (dovek) [inline src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java]: If I don't have the serialization I receive the following error:
"The serializable class  does not declare a static final serialVersionUID field of type long"

**Pull request merged**